cmake/GeneratePkgConfigFiles.cmake

```diff
@@ -23,9 +23,12 @@ macro(GENERATE_PKG_CONFIG_FILES target package_name)
             endforeach()
         endif()
 
+        # Get name for "-l:<name>" linker flags
+        get_target_property(_lib_name ${target} OUTPUT_NAME)
```

Comment:
This will not work in almost all circumstances, as OUTPUT_NAME is mostly calculated by the generator/build processor at build time and thus empty at configure time.
You'll end up with `-l:_lib_name_NOTFOUND` appearing in the .pc file. I actually tried exactly this and it failed in all my configurations. Also tried `OUTPUT_DEBUG_NAME` etc, same issue. It just doesn't work.

Reply:
Hi @kblaschke,

I understand what you are saying, but I cannot confirm it in practice, it works well with generator `Ninja Multi-Config` over here. For proof:

```console
# cd "$(mktemp -d)"
# git clone --depth 1 --branch fix-collisions-with-projectm-prior-to-4-0 https://github.com/hartwork/projectm
# cd projectm
# cmake -S . -B build-ninja-multi -G 'Ninja Multi-Config'
# DESTDIR="$PWD"/ROOT/ ninja -C build-ninja-multi install
# grep -h ^Libs: ROOT/usr/local/lib*/pkgconfig/*.pc
Libs: -L${libdir} -l:projectM-4d  -l/usr/lib/gcc/x86_64-pc-linux-gnu/11/libgomp.so -l/usr/lib64/libpthread.a
Libs: -L${libdir} -l:projectM-4-playlistd
```

What am I missing?

Reply:
If that here works correctly, we can of course keep it that way. Currently having only a Windows machine on hand, so can't test it right now.
Starting after `cd projectm` above:
```bash
# cmake -S . -B build-ninja-multi -G 'Ninja Multi-Config' -DCMAKE_INSTALL_PREFIX="$PWD"/ROOT/
# cmake --build build-ninja-multi --config Debug --parallel
# cmake --build build-ninja-multi --config Release --parallel
# cmake --install build-ninja-multi --config Debug
# cmake --install build-ninja-multi --config Release
```
Then check if both debug and release pkg-config files are installed in the prefix and both have the correct library filenames, e.g. `-l:projectM-4-playlistd` in the projectM-4-playlist-debug.pc file and `-l:projectM-4-playlist` in the projectM-4-playlist.pc file.

Reply:
@kblaschke I believe that^^ is all working now (since today), more details at https://github.com/projectM-visualizer/projectm/pull/699#discussion_r1111125934 .